### PR TITLE
simplify HTTP options

### PR DESCRIPTION
**Pull request by colombod**
Aligning api to comply with issue #444

### CODE REVIEW

File: src/dotnet-interactive.Tests/CommandLine/CommandLineParserTests.cs [hunk 1]
```diff
@@ -192,32 +219,46 @@ public void jupyter_parses_connection_file_path()
         }
 
         [Fact]
-        public async Task jupyter_does_not_enable_http_api_by_default()
+        public async Task jupyter_command_does_enable_http_api_by_default()
```
Comment (jonsequitur) [inline src/dotnet-interactive.Tests/CommandLine/CommandLineParserTests.cs:222]:
```suggestion
        public async Task jupyter_command_enables_http_api_by_default()
```

File: src/dotnet-interactive.Tests/CommandLine/CommandLineParserTests.cs [hunk 2]
```diff
@@ -192,32 +219,46 @@ public void jupyter_parses_connection_file_path()
...
             var result = _parser.Parse($"jupyter --default-kernel bsharp {Path.GetTempFileName()}");
             var binder = new ModelBinder<JupyterOptions>();
             var options = (JupyterOptions)binder.CreateInstance(new BindingContext(result));
+
             options.DefaultKernel.Should().Be("bsharp");
         }
 
         [Fact]
-        public async Task jupyter_returns_error_if_connection_file_path_does_not_exits()
+        public async Task jupyter_command_returns_error_if_connection_file_path_does_not_exits()
```
Comment (jonsequitur) [inline src/dotnet-interactive.Tests/CommandLine/CommandLineParserTests.cs:261]:
```suggestion
        public async Task jupyter_command_returns_error_if_connection_file_path_does_not_exist()
```

File: src/dotnet-interactive/CommandLine/CommandLineParser.cs [hunk 3]
```diff
@@ -291,30 +265,51 @@ Command StdIO()
...
-                        }
-
-                        return startStdIO(
-                            startupOptions,
-                            CreateKernel(options.DefaultKernel, new BrowserFrontendEnvironment(), startupOptions, null),
-                            console);
-                    });
+                    (startupOptions, options, console, context) => startStdIO(
+                        startupOptions,
+                        CreateKernel(options.DefaultKernel, new BrowserFrontendEnvironment(), startupOptions, null),
```
Comment (jonsequitur) [inline src/dotnet-interactive/CommandLine/CommandLineParser.cs:274]: Should we inject the frontend environment?

File: src/dotnet-interactive/CommandLine/readme.md [hunk 4]
```diff
@@ -9,21 +9,21 @@ A number of APIs are available over HTTP in `dotnet-interactive`. You can enable
 dotnet interactive http                                        | Enabled on auto-selected port
 dotnet interactive http --http-port *                          | Enabled on auto-selected port
 dotnet interactive http --http-port 1234                       | Enabled on port 1234
-dotnet interactive http --http-port-range 8001-9000            | Enabled on a port between 8001 and 9000
+dotnet interactive http --http-port-range 8001-9000            | UNSUPPORTED
```
Comment (jonsequitur) [inline src/dotnet-interactive/CommandLine/readme.md:12]: Not sure if you plan was to delete the ones that are unsupported, but I don't think documenting them is necessary.

Comment (colombod) [reply src/dotnet-interactive/CommandLine/readme.md]: removing them

File: src/dotnet-interactive/CommandLine/CommandLineParser.cs (hunk 3 above)
Comment (colombod) [reply src/dotnet-interactive/CommandLine/CommandLineParser.cs:274]: there is no need here to inject, we need to provide one to the create kernel function. That is the composition part for stdio server

**Pull request merged**